static/script/devices/mediaplayer/samsung_streaming.js

```diff
@@ -533,64 +539,100 @@ require.def(
             },
 
             _getClampedTimeForPlayFrom: function (seconds) {
+                if (this._currentPlayer === this.PlayerEmps.StreamingPlayer && !this._updatingTime) {
+                    this._updateRange();
+                }
                 var clampedTime = this._getClampedTime(seconds);
                 if (clampedTime !== seconds) {
                     RuntimeContext.getDevice().getLogger().debug("playFrom " + seconds+ " clamped to " + clampedTime + " - seekable range is { start: " + this._range.start + ", end: " + this._range.end + " }");
                 }
                 return clampedTime;
             },
+            
+            _getClampOffsetFromConfig: function() {
+                var clampOffsetFromEndOfRange;
+                var config = RuntimeContext.getDevice().getConfig();
+                if (config && config.streaming && config.streaming.overrides) {
+                    clampOffsetFromEndOfRange = config.streaming.overrides.clampOffsetFromEndOfRange;
+                }
+
+                if(clampOffsetFromEndOfRange !== undefined) {
+                    return clampOffsetFromEndOfRange;
+                } else if (this._isLiveMedia()) {
+                    return this.CLAMP_OFFSET_FROM_END_OF_LIVE_RANGE;
+                } else {
+                    return this.CLAMP_OFFSET_FROM_END_OF_RANGE;
+                }
+            },
 
             _registerEventHandlers: function() {
                 var self = this;
                 this._playerPlugin.OnEvent = function(eventType, param1, param2) {
 
                     if (eventType !== self.PlayerEventCodes.CURRENT_PLAYBACK_TIME) {
-                        self.logger.info("Received event " + eventType + ' ' + param1);
+                        //self.logger.info("Received event " + eventType + ' ' + param1);
                     }
 
                     switch (eventType) {
 
                         case self.PlayerEventCodes.STREAM_INFO_READY:
-                            self._onMetadata();
+                            self._updateRange()
                             break;
 
                         case self.PlayerEventCodes.CURRENT_PLAYBACK_TIME:
-                            self._onCurrentTime(param1);
-                            //<TODO> use only in LIVE scenario
-                            if (Math.floor(param1/1000) % 8 === 0 && self._state !== MediaPlayer.STATE.STOPPED) {
-                                self._onMetadata();
+                            if (self._range && self._isLiveMedia()) {
+                                var seconds = Math.floor(param1/1000);
+                                //jump to previous current time if PTS out of range occurs
+                                if (seconds > self._range.end + self.RANGE_END_TOLERANCE) {
+                                    self.playFrom(self._currentTime);
```

Comment:
I don't see the necessity of this; we often see current time go beyond the end of the seekable range and have no problem. It's only a problem if we attempt to seek outside the seekable range.